### PR TITLE
Enable Code Formatting Checks 

### PR DESCRIPTION
**Pull request by cssherman**

### CODE REVIEW

**Comment by cssherman** [comment]
@TotoGaz - I'm turning on the yapf code-formatting checks here, and the results for various mesh doctor files look good.  A couple of notes:

1) In one location, I found that implicit tuple assignment could cause problems with the formatter (e.g. `some_tuple = 1, 2`).  Changing this to an explicit assignment worked  (e.g. `some_tuple = (1, 2)`).
2) I added a couple of yapf enable/disable statements to one of the mesh doctor tests that included some structured mesh data.

File: .style.yapf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[style]
+based_on_style = pep8
+spaces_before_comment = 4
```
Comment (TotoGaz) [inline .style.yapf:3]:
```suggestion
spaces_before_comment = 2
```
maybe it's enough?

**Review by TotoGaz — APPROVED** [review]
What about using https://github.com/google/yapf?tab=readme-ov-file#space_inside_brackets ?
I find it more readable + it's like in `geos`.

**Comment by untereiner** [comment]
Hello! The new trend is to use [ruff](https://docs.astral.sh/ruff/): 50x faster than black and 100x than yapf

**Comment by TotoGaz** [comment]
> Hello! The new trend is to use [ruff](https://docs.astral.sh/ruff/) (written in Rust): 50x faster than black and 100x than yapf. Also, would it be possible to migrate to poetry ? (and use pyproject.toml for configurations)

Does it have as many configuration options?
Is it easy to deploy? (since it's in `rust`, maybe it's more complicated to deploy?)

**Comment by untereiner** [comment]
its rust wrapped to python to be fast (as opposed to python :) ) I don't know about the options. It produces pep8 compatible code.
 | yapf  | ruff |
 | :--------------- |:---------------:| 
 | spaces_before_comment | [too-few-spaces-before-inline-comment](https://docs.astral.sh/ruff/rules/too-few-spaces-before-inline-comment/) |
| split_before_logical_operator=true | PEP8  default |
| column_limit | [max-line-length](https://docs.astral.sh/ruff/settings/#pycodestyle-max-line-length) |

**Comment by untereiner** [comment]
I was proposing this change because this style is based on PEP8 (not google or facebook). So not sure yapf has an added value. I was using flake8, isort and black. I am in the process of changing to ruff my projects. So just one tool using a modern build system interface (pyproject.toml)

**Comment by TotoGaz** [comment]
Is the installation easy on Pangea3 for ex? There should be binaries for `ppc`, but I think we should check first if all our targets are properly supported.

**Comment by untereiner** [comment]
very good point @TotoGaz

**Comment by sframba** [comment]
@untereiner the packages seem to be available for ppc, I am trying to install them on Pangea3, I'll keep you posted

**Comment by cssherman** [comment]
@untereiner @TotoGaz - I've played around with ruff and its linter plugins for vscode/sublime, and I'm happy with it.  Assuming that plays well with Pangea3, I'm OK moving forward with it.

**Comment by sframba** [comment]
> @untereiner the packages seem to be available for ppc, I am trying to install them on Pangea3, I'll keep you posted

It seems that the python packages are available for ppc. We don't have any Rust distribution that I know of though. But we can ask the maintenance team or at the very least deploy a local version via Jenkins

**Comment by untereiner** [comment]
I think you do not need any rust distribution installed locally. The code is compiled to the target system without the need of any dependencies.
The documentation says: `pip install ruff`

**Comment by sframba** [comment]
Unfortunately you cannot directly run `pip install` on Pangea3, the installation nodes have no internet connection. You have to manually download the packages on the login node and then install them from the given dir.  

The Rust compiler is required by `cryptography`, which in turn is required by `poetry`. I am currently trying with version `cryptography==3.3.2` which shouldn't need `Rust`. I'll keep you posted

**Comment by sframba** [comment]
So, with `cryptography==3.3.2` does not require `Rust`, so `poetry` is ok. However we have a conflict on `Cython`, since `ruff` requires `Cython~=3.0.0` while our `numpy`/`contourpy` versions require `Cython<3.0.0`. I'll look into it

**Comment by sframba** [comment]
Compilation is ok on Pangea3 with the following versions. Are they ok for you?
```
cryptography==3.3.2
setuptools-rust>=1.7.0
poetry
ruff
Cython==0.29.36
```
Also, 
```
$ ruff -h
<jemalloc>: Unsupported system page size
<jemalloc>: Unsupported system page size
memory allocation of 5 bytes failed
Aborted (core dumped)
```
I suspect this might be an issue. What would you like to test on Pangea3 in particular?

**Comment by untereiner** [comment]
@sframba thank you. This quite extensive installation process demonstrates that ruff isn't adapted to Pangea3. @TotoGaz I was too optimistic.

**Comment by sframba** [comment]
@untereiner let me try to debug the issue on Pangea3 first, it would be a pity to give up just for this. Can someone do the same tests on Lassen?

**Comment by TotoGaz** [comment]
> @sframba thank you. This quite extensive installation process demonstrates that ruff isn't adapted to Pangea3. @TotoGaz I was too optimistic.

Actually, I was quite convince we were going to face an issue, but still it was forth testing first 😉 🤣 

> @untereiner let me try to debug the issue on Pangea3 first, it would be a pity to give up just for this. Can someone do the same tests on Lassen?

@sframba you can do as you want, but TBH I wouldn't waste time on it. We have very good replacement tools. W.r.t. how "critical" this task is in the `geos` ecosystem, I would let it go ❄️

**Comment by cssherman** [comment]
@TotoGaz @untereiner @sframba - let's continue with merging these changes, and plan to re-visit this

File: .style.yapf (hunk 1 above)
Comment (cssherman) [reply .style.yapf]: Trying out the space_inside_brackets setting, I agree that the resulting code looks good.  I'm happy to move forward with it.

**Pull request merged**